pkg/cmd/roachtest/spec/cluster_spec.go

```diff
@@ -99,10 +99,10 @@ const (
 type ClusterSpec struct {
 	Arch      vm.CPUArch // CPU architecture; auto-chosen if left empty
 	NodeCount int
-	// WorkloadNode indicates that the last node of the cluster should be a
-	// workload node. Defaults to a VM with 4 CPUs if not specified by
+	// WorkloadNodes indicates that the last n nodes of the cluster should
```

Comment:
Throwing another idea out there: instead of having users pass in `nodeCount` to `MakeClusterSpec`, there are three new options:
* `CRDBNodes(count int)`
*  `WorkloadNodes(count int)`
* `OtherNodes(label string, count int)`

On `ClusterSpec` itself, you have `NodeCount int` as before and a `map[nodeType]startAndEndIndex`/`map[nodeType]listOfNodes` (exact types are implementation details), with some sentiment label for CRDB and workload nodes. Now whenever an option comes in, you assign it `[NodeCount+1,NodeCount+count)` and then increment `NodeCount`. This would provide a standard interface for users to specify other types of nodes.

Similarly, `NodeLister` would have the following analogous accessors:
* `CRDBNodes()`
* `WorkloadNodes()`
* `OtherNodes(label string)`